test/sql/storage/wal/wal_replay_consistency_for_random.test

```diff
@@ -0,0 +1,51 @@
+# name: test/sql/storage/wal/wal_replay_consistency_for_random.test
+# group: [wal]
+
+statement ok
+PRAGMA disable_checkpoint_on_shutdown;
+
+statement ok
+SET wal_autocheckpoint='1TB'
+
+statement ok
+ATTACH '__TEST_DIR__/wal_replay.db';
+
+statement ok
+USE wal_replay;
+
+statement ok
+CREATE TABLE t AS SELECT range::INT AS id FROM range(10);
+
+statement ok
+ALTER TABLE t ADD COLUMN r DOUBLE DEFAULT RANDOM();
+
+statement ok
+CREATE TABLE original AS SELECT id, r FROM t;
+
+# stop using wal_replay to be able to detach it
+statement ok
+USE memory;
```

Comment:
You're going to run into a test failure here, you can probably add `require noforcestorage` at the top to fix it though